theme.toml

```diff
@@ -64,8 +64,8 @@ label = "honey"
 "ui.selection" = { bg = "#540099" }
 "ui.selection.primary" = { bg = "#540099" }
 # TODO: namespace ui.cursor as ui.selection.cursor?
-"ui.cursor.select" = { bg = "delta" }
-"ui.cursor.insert" = { bg = "white" }
```

Comment:
the old highlight should stay so that the non primary colors should also change mode dependent. You need both the `cursor.primary.*` and `cursor.*` highlight